src/repositories/indexable-hierarchy-repository.php

```diff
@@ -46,9 +67,37 @@ public function add_ancestor( $indexable_id, $ancestor_id, $depth ) {
 		return $hierarchy;
 	}
 
+	/**
+	 * Retrieves the ancestors. Create them when empty.
+	 *
+	 * @param Indexable $indexable The indexable to get the ancestors for.
+	 *
+	 * @return Indexable_Hierarchy[] The ancestors.
+	 */
+	public function get_ancestors( Indexable $indexable ) {
+		$ancestors = $this->query()
+			->where( 'indexable_id', $indexable->id )
+			->order_by_desc( 'depth' )
+			->find_many();
+
+		if ( ! empty( $ancestors ) ) {
+			return $ancestors;
+		}
+
+		$indexable = $this->builder->build( $indexable );
+		$ancestors = $this->query()
+			->where( 'indexable_id', $indexable->id )
+			->order_by_desc( 'depth' )
+			->find_many();
+
+		return $ancestors;
+	}
+
 	/**
 	 * Starts a query for this repository.
 	 *
+	 * @codeCoverageIgnore Too difficult to test because of static. Needs more research.
```

Comment:
Is it? Is mocking the `Yoast_Model::of_type` and checking if that is called with `Indexable_Hierarchy` not enough here?